pkg/envoy/lds/types.go

```diff
@@ -1,9 +1,17 @@
 package lds
 
 import (
+	"github.com/openservicemesh/osm/pkg/catalog"
 	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/service"
 )
 
 var (
 	log = logger.New("envoy/lds")
 )
+
+// listenerBuilder is a type containing data to build the listener configurations
+type listenerBuilder struct {
+	svcAccount  service.K8sServiceAccount
+	meshCatalog catalog.MeshCataloger
```

Comment:
It seems to me that the only reason we need the Mesh Catalog is to get the inbound Service Accounts for an SA via `ListAllowedInboundServiceAccounts()`.  What if instead of passing the mesh catalog reference we actually provided the list of inbound accounts for which we are building rbac policies?

Reply:
This is intentionally not done because the `listenerBuilder` is meant to contain objects useful to build the listener configs in general, not just related to RBAC filter. My intent is to use this beyond just this PR. For ex., to build TCP routes, I will be leveraging the `listenerBuilder`, with more metadata potentially.